Revit_Core_Engine/Convert/Environment/ToRevit/Duct.cs

```diff
@@ -0,0 +1,214 @@
+            if (duct == null)
+                return null;
+
+            // Construct Revit Duct
+            Duct revitDuct = refObjects.GetValue<Duct>(document, duct.BHoM_Guid);
+            if (revitDuct != null)
+                return revitDuct;
+
+            // Settings
+            settings = settings.DefaultIfNull();
+
+            // Duct type
+            DuctType ductType = duct.SectionProperty.ToRevitElementType(document, new List<BuiltInCategory> { BuiltInCategory.OST_DuctSystem }, settings, refObjects) as DuctType;
+
+            // End points
+            XYZ start = duct.StartPoint.ToRevit();
+            XYZ end = duct.EndPoint.ToRevit();
+
+            // Level
+            Level level = document.LevelBelow(Math.Min(start.Z, end.Z), settings);
+            if (level == null)
+                return null;
+
+            // Default system used for now
+            // TODO: in the future you could look for the existing connectors and check if any of them overlaps with start/end of this duct - if so, use it in Duct.Create.
+            // hacky/heavy way of getting all connectors in the link below - however, i would rather filter the connecting elements out by type/bounding box first for performance reasons
+            // https://thebuildingcoder.typepad.com/blog/2010/06/retrieve-mep-elements-and-connectors.html
+            MechanicalSystemType mst = new FilteredElementCollector(document).OfClass(typeof(MechanicalSystemType)).OfType<MechanicalSystemType>().FirstOrDefault();
+            revitDuct = Duct.Create(document, mst.Id, ductType.Id, level.Id, start, end);
+
+            // Copy parameters from BHoM object to Revit element
+            revitDuct.CopyParameters(duct, settings);
+
+            double orientationAngle = duct.OrientationAngle;
+            if (Math.Abs(orientationAngle) > settings.AngleTolerance)
+            {
+                ElementTransformUtils.RotateElement(document, revitDuct.Id, Line.CreateBound(start, end), orientationAngle);
+            }
+
+            SectionProfile sectionProfile = duct.SectionProperty.SectionProfile;
+            if (sectionProfile == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Duct creation requires a SectionProfile. \n No elements created.");
+                return null;
+            }
+
+            DuctSectionProperty ductSectionProperty = duct.SectionProperty;
+
+            //ToDo: Verify builtInParameter for correct value. This is not functioning as expected
+            double flowRate = duct.FlowRate;
+            revitDuct.SetParameter(BuiltInParameter.RBS_DUCT_FLOW_PARAM, flowRate); //Not being set correctly.
+
+            double hydraulicDiameter = ductSectionProperty.HydraulicDiameter;
+            revitDuct.SetParameter(BuiltInParameter.RBS_HYDRAULIC_DIAMETER_PARAM, hydraulicDiameter);
+
+            // Rectangular Duct 
+            if (ductType.Shape == ConnectorProfileType.Rectangular)
```

Comment:
`ductType.Shape` does not compile against Revit 2018 due to missing property in the API - let's catch up and resolve.